tests/kernel/gen_isr_table/src/main.c

```diff
@@ -36,13 +36,22 @@ extern uint32_t _irq_vector_table[];
 /* ARC EMSDP' console will use irq 108 / irq 107, will conflict
  * with isr used here, so add a workaround
  */
-#undef CONFIG_NUM_IRQS
-#define CONFIG_NUM_IRQS 105
+#define TEST_NUM_IRQS	105
+#elif defined(CONFIG_SOC_NRF5340_CPUAPP) || defined(CONFIG_SOC_NRF9160)
+/* In nRF9160 and application core in nRF5340, not all interrupts with highest
+ * numbers are implemented. Thus, limit the number of interrupts reported to
+ * the test, so that it does not try to use some unavailable ones.
+ */
+#define TEST_NUM_IRQS	33
+#else
+#define TEST_NUM_IRQS	CONFIG_NUM_IRQS
 #endif
 
-#define IRQ_LINE(offset)	(CONFIG_NUM_IRQS - ((offset) + 1))
-#define TABLE_INDEX(offset)	(IRQ_TABLE_SIZE - ((offset) + 1))
-#define TRIG_CHECK_SIZE         6
+#define TEST_IRQ_TABLE_SIZE 	(IRQ_TABLE_SIZE - \
+				 (CONFIG_NUM_IRQS - TEST_NUM_IRQS))
+#define IRQ_LINE(offset)	(TEST_NUM_IRQS - ((offset) + 1))
+#define TABLE_INDEX(offset)	(TEST_IRQ_TABLE_SIZE - ((offset) + 1))
```

Comment:
This is not a no-op, @anangl , TABLE_SIZE is now different from what it was before. Are you sure this is not breaking ARC? Or is it actually a fix? If so it should be reflected in the commit msg

Reply:
Hm, OK, I see that IRQ_TABLE_SIZE is defined as 
`#define IRQ_TABLE_SIZE (CONFIG_NUM_IRQS - CONFIG_GEN_IRQ_START_VECTOR)` so it might have been picking the overwrite of CONFIG_NUM_IRQS in the earlier solution (@anangl confirm?)

Reply:
Previously, `CONFIG_NUM_IRQS` was redefined. And this redefinition influenced the value of `IRQ_TABLE_SIZE`, because this macro references `CONFIG_NUM_IRQS` in its [definition](https://github.com/zephyrproject-rtos/zephyr/blob/a2d4292a96e0e3f710374b6da6d3d5060bccc905/include/sw_isr_table.h#L73). I wanted to get rid of this `#undef CONFIG_NUM_IRQS`, and that's why I needed to use `TEST_IRQ_TABLE_SIZE`, which differs from `IRQ_TABLE_SIZE` by the difference between `CONFIG_NUM_IRQS` and `TEST_NUM_IRQS`.

Reply:
> Hm, OK, I see that IRQ_TABLE_SIZE is defined as
> `#define IRQ_TABLE_SIZE (CONFIG_NUM_IRQS - CONFIG_GEN_IRQ_START_VECTOR)` so it might have been picking the overwrite of CONFIG_NUM_IRQS in the earlier solution (@anangl confirm?)

I see that I was writing my comment in parallel with you adding yours. Yes, I confirm.